### PR TITLE
fix(rpc): set const ReadHeaderTimeout to dymint RPC server

### LINKED ISSUE #652: HTTP server misconfiguration allows Slowloris DoS attacks

### PR DESCRIPTION
**Pull request by mtsitrin**
* setting client timeout to mitigate HTTP based DoS attacks

Close #652 

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: rpc/server.go [hunk 1]
```diff
@@ -176,7 +180,10 @@ func (s *Server) startRPC() error {
 
 func (s *Server) serve(listener net.Listener, handler http.Handler) error {
 	s.Logger.Info("serving HTTP", "listen address", listener.Addr())
-	s.server = http.Server{Handler: handler} //#nosec
+	s.server = http.Server{
+		Handler:           handler,
+		ReadHeaderTimeout: ReadHeaderTimeout,
+	}
```
Comment (danwt) [inline rpc/server.go:186]: can we also have readtimeout

And use timeouthandler to wrap the handler?

**Review by danwt — CHANGES_REQUESTED** [review]

**Comment by mtsitrin** [comment]
I've added `ReadHeaderTimeout` to mitigate network based slow attack (which can happen before the execution of `timeouthandler`)

**Comment by danwt** [comment]
refactor :green_apple: 
timeout: we still want https://pkg.go.dev/net/http#TimeoutHandler and  server.ReadTimeout right?

	// ReadTimeout is the maximum duration for reading the entire
	// request, including the body. A zero or negative value means
	// there will be no timeout.
	//
	// Because ReadTimeout does not let Handlers make per-request
	// decisions on each request body's acceptable deadline or
	// upload rate, most users will prefer to use
	// ReadHeaderTimeout. It is valid to use them both.
	ReadTimeout time.Duration
?

**Review by danwt — APPROVED** [review]

**Comment by mtsitrin** [comment]
AFAICT, we want server with `ReadHeaderTimeout` defined and `http.TimeoutHandler` as the handler.

**Review by zale144 — APPROVED** [review]

**Comment by zale144** [comment]
@danwt I think this is about the timeout for reading headers, to prevent someone from sending a ridiculously large header that would take forever to read. But I am not sure if `http.TimeoutHandler` would cover that.

**Pull request merged**